contrib/face_recognition.lua

```diff
@@ -0,0 +1,163 @@
+                        "knownImagePath",
+                        "directory",                                                            -- type
+                        _("Face recognition: Known images"),                                    -- label
+                        _("Path to images with known faces, files named after tag to apply"),   -- tooltip
+                        "~/.config/darktable/face_recognition")                                 -- default
+-- ...and number of CPU cores to use
+dt.preferences.register("FaceRecognition",
+                        "nrCores",
+                        "integer",                                                              -- type
+                        _("Face recognition: Nr of CPU cores"),                                 -- label
+                        _("Number of CPU cores to use, 0 for all"),                             -- tooltip
+                        0,                                                                      -- default
+                        0,                                                                      -- min
+                        64)                                                                     -- max
+                        
+local function show_status (storage, image, format, filename, number, total, high_quality, extra_data)
+  dt.print("Export to Face recognition "..tostring(number).."/"..tostring(total))
+end
+
+local function face_recognition (storage, image_table, extra_data) --finalize
+  if not df.check_if_bin_exists("face_recognition") then
+    dt.print(_("Face recognition not found"))
+    return
+  end
+
+  -- list of exported images
+  local img_list = {}
+
+  for _,v in pairs(image_table) do
+    table.insert (img_list, v)
+  end
+
+  -- Get path of exported images
+  local path = df.get_path (img_list[1])
+  dt.print_error ("FR: Unknown path: " .. path)
+
+  -- Path to known images and number of CPU cores to use
+  local knownPath = dt.preferences.read("FaceRecognition", "knownImagePath", "directory")
+  local nrCores = dt.preferences.read("FaceRecognition", "nrCores", "integer")
+
+  if nrCores < 1 then
+    nrCores = -1
+  end
+    
+  -- Output file
+  local output = path .. "fr.txt"
+  
+  local command = "face_recognition --cpus " .. nrCores .. " " .. knownPath .. " " .. path .. " > " .. output
+  dt.print_error("FR: " .. command)
+  dt.print(_("Starting face recognition..."))
+
+  dt.control.execute(command)
+
+  -- Remove exported images
+  for _,v in ipairs(img_list) do
+    os.remove (v)
+  end
+  
+  -- Open output file
+  local f = io.open(output, "rb")
+  
+  if not f then
+    dt.print(_("Face recognition failed"))
+  else
+    dt.print(_("Face recognition finished"))
+    f:close ()
+  end
+  
+  -- Read output
+  local result = {}
+  for line in io.lines(output) do 
+    local file, tag = string.match (line, "(.*),(.*)$")
+    tag = string.gsub (tag, "%d*$", "")
+    dt.print_error ("F:"..file .." T:".. tag)
```

Comment:
Perhaps it's useful to a little bit more verbose her. "File:" instead of "F:" and "Tag:" instead of "T".